hbase-server/src/test/java/org/apache/hadoop/hbase/master/http/TestMasterHbckServlet.java

```diff
@@ -0,0 +1,237 @@
+    // Set up request and response mocks
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    PrintWriter writer = new PrintWriter(out);
+    doReturn(writer).when(response).getWriter();
+
+    // Set up servlet config
+    ServletContext context = mock(ServletContext.class);
+    Map<String, Object> configMap = new HashMap<>();
+    doAnswer(new Answer<Object>() {
+      @Override
+      public Object answer(InvocationOnMock invocation) throws Throwable {
+        return configMap.get(invocation.getArgument(0));
+      }
+    }).when(context).getAttribute(anyString());
+    doAnswer(new Answer<Void>() {
+      @Override
+      public Void answer(InvocationOnMock invocation) throws Throwable {
+        configMap.put(invocation.getArgument(0), invocation.getArgument(1));
+        return null;
+      }
+    }).when(context).setAttribute(anyString(), any());
+    ServletConfig config = mock(ServletConfig.class);
+    doReturn(context).when(config).getServletContext();
+
+    // Instantiate the servlet and call doGet
+    MasterHbckServlet servlet = new MasterHbckServlet();
+    servlet.init(config);
+    servlet.getServletContext().setAttribute(HMaster.MASTER, master);
+    servlet.doGet(request, response);
+    // Flush the writer
+    writer.flush();
+
+    // Response should be now be in 'out'
+    String responseString = new String(out.toByteArray(), StandardCharsets.UTF_8);
+    Map<String, Object> result =
+      GSON.fromJson(responseString, new TypeToken<Map<String, Object>>() {
+      }.getType());
+
+    // Check that the result is as expected
+    long startTimestamp = ((Double) result.get(MasterHbckServlet.START_TIMESTAMP)).longValue();
+    assertEquals(FAKE_START_TIMESTAMP, startTimestamp);
+    long endTimestamp = ((Double) result.get(MasterHbckServlet.END_TIMESTAMP)).longValue();
+    assertEquals(FAKE_END_TIMESTAMP, endTimestamp);
+    Map<String, Object> inconsistentRegions =
+      (Map<String, Object>) result.get(MasterHbckServlet.INCONSISTENT_REGIONS);
+    assertNotNull(inconsistentRegions);
+    assertEquals(1, inconsistentRegions.size());
+    assertNotNull(inconsistentRegions.get(FAKE_HRI.getEncodedName()));
+    assertNull(inconsistentRegions.get(FAKE_HRI_3.getEncodedName()));
+    Map<String, Object> orphanRegionsOnRS =
+      (Map<String, Object>) result.get(MasterHbckServlet.ORPHAN_REGIONS_ON_RS);
+    assertNull(orphanRegionsOnRS);
+    Map<String, String> orphanRegionsOnFS =
+      (Map<String, String>) result.get(MasterHbckServlet.ORPHAN_REGIONS_ON_FS);
+    assertNotNull(orphanRegionsOnFS);
+    assertEquals(1, orphanRegionsOnFS.size());
+    assertNull(orphanRegionsOnFS.get(FAKE_HRI.getEncodedName()));
+    assertNotNull(orphanRegionsOnFS.get(FAKE_HRI_3.getEncodedName()));
+    List holes = (List) result.get(MasterHbckServlet.HOLES);
```

Comment:
Given that we are just comparing null vs non-null and size of the list, this raw use of `List` is fine.